docs/kpm_sparse_checkout.md

```diff
@@ -0,0 +1,75 @@
+## KPM Sparse-Checkout feature
+
+## 1. Introduction
+kpm which is the package management tool for KCL, does not support sparse-checkout, which means an issue arises when dealing with monorepos, which may contain many KCL packages. We need to get a solution to download specific packages instead of all in a monorepo.
+
+## 2. User Worklow
+- Adding a Subdirectory Dependency via Command Line
+- Specifying a Subdirectory in `kcl.mod`
+- Adding a Subdirectory Dependency via Configuration File 
+
+## 3. Command Line Inteface 
+We will try to keep the command line interface as simple and straightforward as possible for adding a Git repository subdirectory as a dependency. The below steps show how a user can use the feature:
+
+- Adding a Dependency :
+The following command will lead to the addition of a dependency:
+`kcl mod add <repository_url> <subdirectory_path>`
+
+This command will add the specified subdirectory as a dependency in the current KCL project.
+
+- Non-interactive Method :
+In addition to the command line interface, users can specify subdirectory dependencies directly in the `kcl.mod` file, which can be particularly useful for automated environments.
+
+## 4. Example Use-case
+Considering the nginx-ingres module, on typing the command 
+
+```
+kcl mod add https://github.com/kcl-lang/modules/tree/main/nginx-ingress /restrict-ingress-anotations
```

Comment:
I just have a question, why don't you just merge these two parameters, something like this? 
```
kcl mod add https://github.com/kcl-lang/modules/tree/main/nginx-ingress/restrict-ingress-anotations
```
I'm not sure what the problems will be behind such a merging, so you need to consider it or have a research.

Reply:
Done the changes, i did some research and gave thought to it but i don't think there won't be much of any problem. If any problems arise during implementation, i'll adjust accordingly.